src/main/scala/chisel3/util/Decoupled.scala

```diff
@@ -239,8 +244,21 @@ class Queue[T <: Data](val gen: T,
     maybe_full := do_enq
   }
 
+  if (hasFlush) {
+  // when flush is on, empty the queue
+    when(io.flush.get) {
+      enq_ptr.reset()
+      deq_ptr.reset()
+    }
+
+    when(io.flush.get) {
+      maybe_full := false.B
+    }
+  }
+
```

Comment:
```suggestion
  io.flush.foreach { flush => 
  // when flush is on, empty the queue
    when(flush) {
      enq_ptr.reset()
      deq_ptr.reset()
      maybe_full := false.B
    }
  }

```

Reply:
or, do you even need to conditionalize this on the `io.flush.foreach`? You already have a well-defined `flush` signal...